### PR TITLE
Added a "Restart from Checkpoint" button

### PR DESCRIPTION
**Pull request by HybridDog**
Original PR: #885

I've changed it so that the kill function is executed, which plays the hurt sound (otherwise the player would silently loose the coins).

### CODE REVIEW

File: src/supertux/player_status.hpp [hunk 1]
```diff
@@ -40,11 +40,13 @@ class PlayerStatus final
   PlayerStatus();
   void reset();
   void add_coins(int count, bool play_sound = true);
+  void take_checkpoint_coins() { coins -= std::max(coins / 10, 25); }
```
Comment (tobbi) [inline src/supertux/player_status.hpp]: AppVeyor complains about
```
'max': identifier not found
```

Comment (HybridDog) [reply src/supertux/player_status.hpp]: It cannot find std::max `error C2039: 'max': is not a member of 'std'`. Maybe it's an unreliable transitive header inclusion.
https://ci.appveyor.com/project/supertux/supertux-o8t59/builds/25955016#L5821

**Comment by tobbi** [comment]
I assume that the remaining build issues require a cache purge, so nothing that's caused by this PR. Thanks 👍

**Comment by Alzter** [comment]
Awesome, but change the text to just "Restart from Checkpoint" to avoid padding out the pause menu.

**Comment by HybridDog** [comment]
> Awesome, but change the text to just "Restart from Checkpoint" to avoid padding out the pause menu.

I think this only works as long as the message is not translated to a long one.

**Pull request merged**